cmd/yggdrasil/main.go

```diff
@@ -272,6 +272,10 @@ func main() {
 		}
 	}
 
+	if _, err = notifyStartupCompleted(); err != nil {
+		log.Warnln("Error while sending start up notification:", err)
+	}
+
```

Comment:
There is no guarantee at this point (in the main goroutine) that startup tasks in other goroutines have been finished, since they don't synchronise.

So I think you are just racing against tun(4) setup just like `-chuser` is already racing against the control socket setup over at https://github.com/yggdrasil-network/yggdrasil-go/pull/927#discussion_r1826606055